File: ga/glm4_result_2024_6/result/1968E_cpp_glm/2_1968E.cpp
use std::io;

fn solve() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Invalid input");

    println!("1 1");
    println!("2 1");

    let mut t = n - 2;
    if t > 0 {
        println!("{} {}", n, n);
        t -= 1;
    }

    for i in 3..=n {
        if t > 0 {
            println!("1 {}", i);
            t -= 1;
        }
    }
}

fn main() {
    let mut t = 1;
    loop {
        solve();
        println!(""); // Print newline between test cases

        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        t = match input.trim().parse::<usize>() {
            Ok(0) => break,
            Ok(num) => num,
            Err(_) => continue,
        };
    }
}

File: ga/4omini_result_2024_7/result/1968E_cpp_mini/2_1968E.rs
use std::io;

fn solve(n: usize) {
    let mut t = n;
    println!("1 1");
    println!("2 1");
    t -= 2;
    
    if t > 0 {
        println!("{} {}", n, n);
        t -= 1;
    }
    
    for i in 3..3 + t {
        println!("1 {}", i);
    }
}

fn main() {
    let stdin = io::stdin();
    let mut input = String::new();
    stdin.read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        input.clear(); // Clear the input buffer for the next testcase
        stdin.read_line(&mut input).unwrap();
        let n: usize = input.trim().parse().unwrap();
        solve(n);
    }
}